cluster/kubernetes/resource/fluxhelmrelease.go

```diff
@@ -220,18 +327,109 @@ func interpretAsImage(m mapper) (image.Ref, ImageSetter, bool) {
 	return image.Ref{}, nil, false
 }
 
+// containerImageMappingsFromAnnotations collects yaml dot notation
+// mappings of container images from the given annotations.
+func containerImageMappingsFromAnnotations(annotations map[string]string) map[string]ContainerImageMap {
+	cim := make(map[string]ContainerImageMap)
+	for k, v := range annotations {
+		switch {
+		case strings.HasPrefix(k, ImageRegistryPrefix):
+			container := strings.TrimPrefix(k, ImageRegistryPrefix)
+			i, _ := cim[container]
+			i.Registry = v
+			cim[container] = i
+		case strings.HasPrefix(k, ImageRepositoryPrefix):
+			container := strings.TrimPrefix(k, ImageRepositoryPrefix)
+			i, _ := cim[container]
+			i.Repository = v
+			cim[container] = i
+		case strings.HasPrefix(k, ImageTagPrefix):
+			container := strings.TrimPrefix(k, ImageTagPrefix)
+			i, _ := cim[container]
+			i.Tag = v
+			cim[container] = i
+		}
+	}
+	for k, _ := range cim {
+		i, _ := cim[k]
+		i.BasePath = ImageBasePath
+		cim[k] = i
+	}
+	return cim
+}
+
+func interpretMappedContainerImage(values map[string]interface{}, cim ContainerImageMap) (image.Ref, ImageSetter, bool) {
+	v, err := gabs.Consume(values)
+	if err != nil {
+		return image.Ref{}, nil, false
+	}
+
+	imageValue := v.Path(cim.Repository).Data()
+	if img, ok := imageValue.(string); ok {
+		if cim.Registry == "" && cim.Tag == "" {
+			if imgRef, err := image.ParseRef(img); err == nil {
+				return imgRef, func(ref image.Ref) {
+					v.SetP(ref.String(), cim.Repository)
+				}, true
+			}
```

Comment:
If this fails, I'm not sure it's worth continuing -- adding a registry and/or tag to an invalid image can't make it valid.

Reply:
(it does get checked again later, so it's safe as it is I believe)